src/common/commonutils/FileUtils.c

```diff
@@ -120,25 +120,52 @@ bool SavePayloadToFile(const char* fileName, const char* payload, const int payl
 
 bool AppendPayloadToFile(const char* fileName, const char* payload, const int payloadSizeBytes, void* log)
 {
-    char* fileContents = NULL;
+    FILE* existingFile = NULL;
+    struct stat st;
     bool result = false;
+    char lastCharacter = EOF;
+
+    if(NULL == fileName || NULL == payload || 0 == payloadSizeBytes)

```

Comment:
>if(NULL == fileName || NULL == payload || 0 == payloadSizeBytes) [](http://example.com/codeflow?start=4&length=64)

Here and on line 135, add inner parenthesis. Never rely on just operator precedence.